src/main/java/ratpack/zipkin/internal/RatpackCurrentTraceContext.java

```diff
@@ -112,10 +116,15 @@ private TraceContextHolder(final TraceContext context) {
 
     @Override
     public void init(Execution execution) {
-      execution
-        .maybeParent()
+      Optional<ExecutionRef> maybeParent =  execution.maybeParent();
+
+      maybeParent
         .flatMap(parent -> parent.maybeGet(TraceContextHolder.class))
         .ifPresent(execution::add);
+
+      maybeParent
+          .flatMap(parent -> parent.maybeGet(DefaultClientTracingInterceptor.ClientSpanHolder.class))
```

Comment:
Copying the `ClientSpanHolder` forward on forked executions as the key to making the client interceptor work.  Effectively the client interceptor request and receive have different executions.  Fortunately the receive is a child of the request at runtime and copying forward the `ClientSpanHolder` saved the day.

Reply:
add comment so tomorrow is saved, too? :D